### PR TITLE
Update the docs for GitHub sync

### PR DESCRIPTION
**Pull request by laurakalbag**
(Don’t merge until Sync is live!)

- Renamed “Import from GitHub” to “Connect GitHub repo”
- Added basic docs on the feature and permissions

### CODE REVIEW

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:2]: I may have missed a conversation about this but is this like a "Docs title" for the feature as opposed to "GitHub Sync" we use elsewhere?

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:10]: Adding this comment here because this paragraph links to the Stately Studio Pro Plan page. Should we also include updating "GitHub Sync (coming soon!)" by removing the "(coming soon!)"?

Comment (kevinmaes) [reply docs/import-from-github.mdx:10]: And I guess that Pro feature bullet point can then link back to this page as well.

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:36]: Since there's just "Create pull request" and "Update pull request" should we make both the same level with `###` The spacing between them is less when one is nested but it looks slightly off with only one child. No strong opinion here though.

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:18]: I love the clarity of just 3 simple steps to get started here and notably they align with the "Steps 1/3" in the modal UI for the feature. However, I didn't see where we actually mention "Connect GitHub repo" as a button or where to find that button. Do we want to start with something like this to kick it off?

"Upon clicking the Connect GitHub repo button, just above your projects on the My Projects or Team Projects pages, there are three steps to setting up your GitHub repo as a connected project."

If that's too long, we can cut out some words.

"On the My Projects or Team Projects pages, click the Connect GitHub repo button and follow these three steps to set up your GitHub repo as a connected project."

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:26]: TODO in the near future would be to expand on how not selecting auto-sync works. This is one of the "beta" features that we're still refining before we go into detail here. For now, we'll encourage keeping autoSync turned on, by default.

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:30]:
```suggestion
You can choose a **Path for new files** for when you create new machines in this connected project. These machines will be added at this relative path from the root of your repo. The default path is **src/stately-studio** and will be created in your repo the firs time you add new machines from Studio to a pull request.
```

@mellson Can you confirm that this is accurate, that it only adds this folder when needed? Otherwise, I can test it out tomorrow.

File: docs/import-from-github.mdx
Comment (kevinmaes) [inline docs/import-from-github.mdx:46]: Here's a thought that's not exclusive to this paragraph but it would probably help to refer to the modal UI that is opened from "repository name in the footer of the left drawer" as "GitHub Settings". That's what the menu item is named in Editor Menu > Project > GitHub Settings where it opens the same modal. If we start to use that terminology it'll be easier to document its contents from now on. We don't want to be stuck having to refer to the repo name in the footer of the left drawer if we can train users to think "GitHub Settings" after they learn how to open that from both locations (footer and Editor Menu).

File: docs/import-from-github.mdx
Comment (kevinmaes) [reply docs/import-from-github.mdx:2]: After reading the rest of the p age I see this is really the button's name so it makes more sense as a lead in. Anyway, I was just wondering because it stood out (3 words longer, bold) in the left sidebar.

**Review by kevinmaes — CHANGES_REQUESTED** [review]
@laurakalbag This is awesome. I think you managed to capture all of the aspects of this MVP release in great detail but in a way that reads clearly and concisely. I left a few ideas, some totally optional, but even if it were to go out as it it would be great.

File: docs/import-from-github.mdx
Comment (laurakalbag) [reply docs/import-from-github.mdx:2]: @kevinmaes it’s because it’s not named GitHub sync anywhere in the interface. This is a bit like the Stately Sky situation, where folks didn’t know where to find it because while we were calling it Stately Sky publicly, nowhere in the Studio said Stately Sky. I’m trying to avoid that problem by making the docs naming follow the UI naming!

File: docs/import-from-github.mdx
Comment (mellson) [reply docs/import-from-github.mdx:30]: Correct: It only adds this on demand. If the folder doesn't already exist, it will be added when the first pull request is created in the Studio.

Comment (kevinmaes) [reply docs/import-from-github.mdx:30]: @laurakalbag I have a typo in my suggestion, "firs time". Sorry.

File: docs/import-from-github.mdx
Comment (kevinmaes) [reply docs/import-from-github.mdx:2]: That sounds good!

**Comment by kevinmaes** [comment]
@laurakalbag GitHub Sync has been released to Prod now for all paid subscribers so we can finish this up any time now. Let me know if you'd like any help.

**Comment by kevinmaes** [comment]
> @laurakalbag GitHub Sync has been released to Prod now for all paid subscribers so we can finish this up any time now. Let me know if you'd like any help.

We had to roll back the release so GitHub Sync is not yet released after all.

**Review by mellson — APPROVED** [review]
Thank you, Laura. This is really good! And I think it's ready to be merged in. There's a detail regarding whether new machines will be created using XState v4 or v5 that we might want to add at some point. Feel free to add it now or let me know if you want me to create a new PR with that information.

CleanShot 2024-01-03 at 07 24 41@2x

File: docs/import-from-github.mdx
Comment (laurakalbag) [reply docs/import-from-github.mdx:10]: Yes, thanks! I’ll do this.

File: docs/import-from-github.mdx
Comment (laurakalbag) [reply docs/import-from-github.mdx:36]: I made it a subheading because it’s supplementary to the information about pull requests (because you need to have made a pull request to update a pull request)

File: docs/import-from-github.mdx
Comment (laurakalbag) [reply docs/import-from-github.mdx:52]: @kevinmaes I’m wary of using names when they’re not visible in the UI. Right now there’s no button that says “GitHub settings”, so as frustratingly verbose as the explanation is, it tells them where to find it.

**Comment by laurakalbag** [comment]
@mellson that’s really useful to know, thanks! I’ve added a tip box on the page with this info. Will merge when I’ve got approval from @kevinmaes on the requested changes.

**Pull request merged**
